### PR TITLE
[Bugfix] Make TODO parser run on develop branch and test source

### PR DESCRIPTION
**Pull request by jwwwb**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
Briefly describe the changes you have introduced.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -48,7 +48,7 @@ include = [
 zenml = "zenml.cli.cli:cli"
 
 [tool.poetry.dependencies]
-python = ">=3.6.2,<3.9"
+python = ">=3.8.0,<3.9"
```
Comment (schustmi) [inline pyproject.toml:51]: Not sure if this is a change you made or that's coming from the fuckup of all PR's, but we probably want this to stay python 3.6 for now right?

**Review by schustmi — CHANGES_REQUESTED** [review]
LGTM, just one small comment

File: pyproject.toml (hunk 1 above)
Comment (jwwwb) [reply pyproject.toml:51]: oh crap you are ri ght, that snuck ino to my commit. will fix it

File: tests/unit/scripts/test_update_todos.py [hunk 2]
```diff
@@ -41,24 +41,23 @@ def test_todo_detection(tmp_path):
 
     file.write_text(
         """
-            # TODO [LOW]: valid todo
+            # TODO [ENG-345]: valid todo
```
Comment (schustmi) [inline tests/unit/scripts/test_update_todos.py:44]: Oh shit I just detected another issue: This is a test file to test the todo script, which now got modified as it's in the tests directory :D 
We probably have to hard code some exclude for this, and definitely close the created jira issues

Comment (jwwwb) [reply tests/unit/scripts/test_update_todos.py:44]: I just added the option to put a `# notodo` flag in a file, inspired by `noqa`.... just have to write a test for it – I didn't even realize there was a test suite for the script, else I would have adjusted that anyway.

Comment (schustmi) [reply tests/unit/scripts/test_update_todos.py:44]: Love the `# notodo` tag! Yeah I completely forgot as well that we might have some issues with that file being in there

**Review by htahir1 — APPROVED** [review]
This LGTM -> We should delete the excess TODO's created from JIRA

Really like the notodo feature!

File: tests/unit/scripts/test_update_todos.py [hunk 3]
```diff
@@ -33,6 +33,34 @@ def setup_environment():
     os.environ["GITHUB_SHA"] = ""
 
 
+def test_no_todo_flag(tmp_path):
```
Comment (schustmi) [inline tests/unit/scripts/test_update_todos.py:36]: I think we're still missing a `# notodo` in this file to exclude it right?

Comment (jwwwb) [reply tests/unit/scripts/test_update_todos.py:36]: good point, fixed

**Review by schustmi — APPROVED** [review]

**Pull request merged**